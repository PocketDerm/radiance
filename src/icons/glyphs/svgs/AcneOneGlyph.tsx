import * as React from 'react';

function SvgAcneOneGlyph(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      viewBox="0 0 48 48"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M23.995 26A1.996 1.996 0 0122 23.995c0-1.107.897-1.995 1.995-1.995S26 22.897 26 23.995A2.017 2.017 0 0123.995 26zm0-2.807a.8.8 0 00-.802.802c0 .449.363.812.802.812.44 0 .812-.363.812-.812a.821.821 0 00-.812-.802z"
        fill="currentColor"
      />
    </svg>
  );
}

export default SvgAcneOneGlyph;
