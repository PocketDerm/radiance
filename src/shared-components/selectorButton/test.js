import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AcneOneGlyph from '../../svgs/glyphs/acne-one-glyph.svg';

import SelectorButton from './index';

describe('<SelectorButton />', () => {
  describe('UI snapshots', () => {
    test('when children is undefined', () => {
      const tree = renderer
        .create(<SelectorButton checked={false} onClick={() => {}} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when children is a node', () => {
      const tree = renderer
        .create(
          <SelectorButton checked={false} onClick={() => {}}>
            SelectorButton Text
          </SelectorButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when checked type is primary', () => {
      const tree = renderer
        .create(
          <SelectorButton checked onClick={() => {}} type="primary">
            SelectorButton Text
          </SelectorButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when checked type is secondary', () => {
      const tree = renderer
        .create(
          <SelectorButton checked onClick={() => {}} type="secondary">
            SelectorButton Text
          </SelectorButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when is checkbox', () => {
      const tree = renderer
        .create(
          <SelectorButton
            checked={false}
            onClick={() => {}}
            selector="checkbox"
          >
            SelectorButton Text
          </SelectorButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    describe('when Icon added', () => {
      it("hides icon for checkbox with size 'small'", () => {
        const tree = renderer.create(
          <SelectorButton
            checked={false}
            onClick={() => {}}
            selector="checkbox"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="small"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });

      it("hides icon for radio button size 'small'", () => {
        const tree = renderer.create(
          <SelectorButton
            checked={false}
            onClick={() => {}}
            selector="radio"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="small"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });

      it("displays icon for checkbox with size 'large'", () => {
        const tree = renderer.create(
          <SelectorButton
            checked={false}
            onClick={() => {}}
            selector="checkbox"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });

      it("displays icon for radio button with size 'large'", () => {
        const tree = renderer.create(
          <SelectorButton
            checked={false}
            onClick={() => {}}
            selector="radio"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });

      it('displays check mark for checked checkbox', () => {
        const tree = renderer.create(
          <SelectorButton
            checked
            onClick={() => {}}
            selector="checkbox"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });

      it('displays check mark for checked radio button', () => {
        const tree = renderer.create(
          <SelectorButton
            checked
            onClick={() => {}}
            selector="radio"
            icon={<AcneOneGlyph width={16} height={16} />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('onClick callback', () => {
    it('is invoked on click', () => {
      const spy = jest.fn();
      const wrapper = shallow(<SelectorButton checked={false} onClick={spy} />);

      wrapper.simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });
});
