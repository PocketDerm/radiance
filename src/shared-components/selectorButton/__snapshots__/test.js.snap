// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectorButton /> UI snapshots when checked type is primary 1`] = `
.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-7:focus {
  outline: none;
}

.emotion-7:focus .emotion-4 {
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
  border-radius: 100%;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-5 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

.emotion-0 {
  color: #ffffff;
  cursor: pointer;
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  width: 2rem;
  border-radius: 100%;
  background-color: #332e54;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={true}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when checked type is secondary 1`] = `
.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-7:focus {
  outline: none;
}

.emotion-7:focus .emotion-4 {
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
  border-radius: 100%;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-5 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

.emotion-0 {
  color: #ffffff;
  cursor: pointer;
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  width: 2rem;
  border-radius: 100%;
  background-color: #a6a1e2;
  border-color: #a6a1e2;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={true}
      className="emotion-1 emotion-2"
      type="secondary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when children is a node 1`] = `
.emotion-0 {
  color: transparent;
  cursor: pointer;
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-7:focus {
  outline: none;
}

.emotion-7:focus .emotion-4 {
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
  border-radius: 100%;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  width: 2rem;
  border-radius: 100%;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

.emotion-5 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when children is undefined 1`] = `
.emotion-0 {
  color: transparent;
  cursor: pointer;
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-5:focus {
  outline: none;
}

.emotion-5:focus .emotion-4 {
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
  border-radius: 100%;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  width: 2rem;
  border-radius: 100%;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-5 emotion-6"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
</div>
`;

exports[`<SelectorButton /> UI snapshots when is checkbox 1`] = `
.emotion-0 {
  color: transparent;
  cursor: pointer;
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-5 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-7:focus {
  outline: none;
}

.emotion-7:focus .emotion-4 {
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
  border-radius: 4px;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  width: 2rem;
  border-radius: 4px;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    SelectorButton Text
  </p>
</div>
`;
